File: keystone-frontend/src/types/workOrder.ts
export type WorkOrder = {

    id: number;

    workOrderNumber: string;

    title: string;

    description: string;

    priority: "Low" | "Medium" | "High" | "Critical";

    status:
        | "New"
        | "Assigned"
        | "In Progress"
        | "Completed"
        | "Closed";

    scheduledDate: string;

    completedDate?: string;

    estimatedDuration: number;

    attachment?: string;

    customerId: number;

    siteId: number;

    technicianId: number;

};

File: FieldServiceMgmtSystem/src/main/java/com/keystone/FieldServiceMgmtSystem/entity/WorkOrder.java
package com.keystone.FieldServiceMgmtSystem.entity;

import java.time.LocalDate;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "work_orders")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class WorkOrder {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String workOrderNumber;

    @Column(nullable = false)
    private String title;

    @Column(length = 1000)
    private String description;

    @Column(nullable = false)
    private String priority;

    @Column(nullable = false)
    private String status;

    @Column(nullable = false)
    private LocalDate scheduledDate;

    private LocalDate completedDate;

    @Column(nullable = false)
    private Integer estimatedDuration;

    private String attachment;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "customer_id", nullable = false)
    private Customer customer;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "site_id", nullable = false)
    private Site site;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "technician_id", nullable = false)
    private Technician technician;
}